starter-code/app.js

```diff
@@ -1,86 +1,106 @@
-var express = require('express');
-var path = require('path');
-var favicon = require('serve-favicon');
-var logger = require('morgan');
-var cookieParser = require('cookie-parser');
-var bodyParser = require('body-parser');
-var app = express();
-
-var index = require('./routes/index');
-var users = require('./routes/users');
-const passportRouter = require("./routes/passportRouter");
-//mongoose configuration
-const mongoose = require("mongoose");
-mongoose.connect("mongodb://localhost/passport-local");
-//require the user model
-const User = require("./models/user");
-const session       = require("express-session");
-const bcrypt        = require("bcrypt");
-const passport      = require("passport");
-const LocalStrategy = require("passport-local").Strategy;
-const flash = require("connect-flash");
-
-
-
-
-
-//enable sessions here
-
-
-
-
-//initialize passport and session here
-
-
-
-
+const express = require('express')
+const path = require('path')
+const favicon = require('serve-favicon')
+const logger = require('morgan')
+const cookieParser = require('cookie-parser')
+const bodyParser = require('body-parser')
+const app = express()
+
+const index = require('./routes/index')
+const users = require('./routes/users')
+const passportRouter = require('./routes/passportRouter')
+
+// mongoose configuration
+const mongoose = require('mongoose')
+mongoose.connect('mongodb://localhost/passport-local')
+
+// require the user model
+const User = require('./models/user')
+
+const session = require('express-session')
+const bcrypt = require('bcrypt')
+const passport = require('passport')
+const LocalStrategy = require('passport-local').Strategy
+
+const flash = require('connect-flash')
+
+const authRoutes = require('./routes/passportRouter')
+app.use('/', authRoutes)
+
+// enable sessions here
+
+app.use(session({
+  secret: 'our-passport-local-strategy-app',
+  resave: true,
+  saveUninitialized: true
+}))
```

Comment:
The session setup must be after the cookie parser. Else it will create a new session everytime you are doing a request.